lib/ASN1/Base128.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+namespace FG\ASN1;
+
+/**
+ * A base-128 decoder.
+ */
+class Base128
+{
+    /**
+     * @param integer $value
+     *
+     * @return string
+     */
+    public static function encode($value)
+    {
+        $octets = chr($value & 0x7F);
+        $value >>= 7;
+
+        while ($value > 0) {
+            $octets .= chr(0x80 | ($value & 0x7F));
+            $value >>= 7;
+        }
+
+        return strrev($octets);
```

Comment:
X.690 section 8.1.2.4.2 b) states the following about the encoding of the long form identifier:
> bits 7 to 1 of the first subsequent octet, followed by bits 7 to 1 of the second subsequent octet, followed in turn by bits 7 to 1 of each further octet, up to and including the last subsequent octet in the identifier octets shall be the encoding of an unsigned  binary integer equal to the tag number, with bit 7 of the first subsequent octet as the most significant bit;

This describes the octets following the first identifier octet which holds, the class, primitive/constructed, and long form marker. The important part here is at the very end of the paragraph: *with bit 7 of the first subsequent octet as the most significant bit*.

The way I understand this I would do the encoding as follows:

```
value:   0x80
binary:  10000000
septets:  1000000  0000000
padded:  11000000 00000000
hex:     0xC0     0x00
```

However this is not what is happening in your implementation and I am a bit uncertain about my interpretation of the standard and would be happy about your input.

I tried to come up with an alternative implementation but I just dont have enough time currently. Maybe I can finish it over the weekend but I can't promise anything. If I do I will push this to a separate branch and wil defer the merge until its completely clear to me how the encoding needs to be implemented.